changelog.d/7421.misc

```diff
@@ -0,0 +1 @@
+Move logs about "Discarding RDATA for unconnected stream federation" to debug.
```

Comment:
suggest we make this match the changelog for the PR that introduced it

Reply:
Good point, fixed in https://github.com/matrix-org/synapse/pull/7421/commits/9c8c3f5625264467a0761c934cdf287818c281c2